knowledge/data-sources/onedrive/main.go

```diff
@@ -209,8 +210,15 @@ func writeMetadata(ctx context.Context, output *MetadataOutput, gptscript *gptsc
 	return gptscript.WriteFileInWorkspace(ctx, ".metadata.json", data)
 }
 
-func getChildrenFileForItem(ctx context.Context, client *msgraphsdk.GraphServiceClient, item models.DriveItemable) ([]models.DriveItemable, error) {
+func syncChildrenFileForItem(ctx context.Context, client *msgraphsdk.GraphServiceClient, gptscriptClient *gptscript.GPTScript, item models.DriveItemable, output *MetadataOutput, root string, logErr *logrus.Logger) ([]models.DriveItemable, error) {
 	if item.GetFile() != nil {
+		// We only sync item that is less than 50 MB, as most of the bigger files won't be supported from knowledge
```

Comment:
Should we rather add a check if the filetype is likely to be supported? Does OneDrive give a mimetype for files?